### DISCUSSION TITLE
How does Unleash parse UserId?

### DISCUSSION THREAD

**Discussion by Thomas-Rice** [Unleash/unleash#2351] [General]
Hi, 
I have a general question. I'm looking at using unleash & was wondering how userIds are interpreted? Does unleash look at headers / cookies etc or are userIds passed in another way?

**Comment by sighphyre** [comment]
Hey @Thomas-Rice!

So it depends a little on what you mean by that. It sounds like you're asking about how the SDKs handle userIds (please shout if I'm wrong about that!).

It depends on which type of [SDK](https://docs.getunleash.io/sdks) you're using, for the server side SDKs, the userId is passed into the SDK itself as part of the [Context](https://docs.getunleash.io/user_guide/unleash_context), in that case, the userId never leaves your servers, everything is evaluated locally. 

For the Client Side SDKs, they're also part of the context property, but that context is destructed into a set of URL parameters, you see see a sample in action [here](https://docs.getunleash.io/user_guide/proxy-api#the-proxy-api). So, effectively userId is passed as a URL param.

Did that answer your question?

**Reply by Thomas-Rice** [reply]
Hi @sighphyre 
Thank you very much for the information! 
I believe I understand, it's passing in UserIDs via parameters/values via url/json :)

**Reply by sighphyre** [reply]
Nothing but a pleasure, happy Unleashing
